### PR TITLE
Add glossary.md

### PR DESCRIPTION
**Pull request by ImJasonH**
This briefly describes some terms commonly thrown around when discussing Knative development and usage, and can serve as a handy reference guide when talking to new users and developers.

The list of terms is faaaar from exhaustive, these are just the few that came up in the last Build WG meeting, which we thought might be useful to document.

### CODE REVIEW

**Comment by samodell** [comment]
Solves Issue #170

**Comment by samodell** [comment]
After taking a look, I think this is a bit different than what we were imagining with issue #170 . It might make more sense to call this a "tools list" or something else other than "glossary," since we're not really defining terms here so much as explaining what tools are helpful when working with Knative.

What do you think about renaming this from "glossary" to something else, @ImJasonH ?

**Comment by ImJasonH** [comment]
I don't have a strong affinity to the name, I just can't think of any better alternatives. :smile: 

For now these are all tools (except `go-containerregistry`, which is a _library_ used by some tools), but I suspected this would expand to include other things over time.

#170 seems to describe a glossary of Knative's own terms (Configuration, Route, Build, etc.), which definitely seems useful, but I agree probably doesn't belong in the same list alongside `ko` and `skaffold`.

If you have a better name in mind, I'm happy to change it, I just don't know to what. :thinking:

**Comment by samodell** [comment]
Hmm... something like "helpful technologies" or "related tools and technologies"? We definitely will have a glossary of terms sometime in the near future, so it'd be great to reserve glossary.md for that!

File: resources.md [hunk 1]
```diff
@@ -1,11 +1,77 @@
-# Glossary
+# Resources
 
-There is a dizzying array of tools used by Knative, and during Knative
-development, and it can be confusing to keep them all straight. This is an
-attempt to briefly document some terms that you might hear while working with
-or using Knative.
+This page contains informations about various tools and technologies
```
Comment (ImJasonH) [inline resources.md]: Typo: information

File: resources.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+that are useful to anyone developing on Knative.
+
+## Community Resources
+
+This section contains tools and technologies developed by members of the
+Knative community specifically for use with Knative.
+
+### [`knctl`](https://github.com/cppforlife/knctl)
+
+`knctl` is an under-development command line tool for working with
```
Comment (ImJasonH) [inline resources.md]: Link to more information? Otherwise if there isn't any more than this I think we should remove it until there is more info

Comment (samodell) [reply resources.md]: There is a link to the knctl repo, and there's some documentation there about how to work with it. What other information would you want included here? I could add that it's an alternative to working directly with kubectl.

Comment (ImJasonH) [reply resources.md]: Each of the other sections has at least a couple explanatory/introductory paragraphs, even though you could link to each repo and just tell people to read there.

It's not a big deal, it just feels light compared to the others.

**Pull request merged**